pytests/test_dct.py

```diff
@@ -0,0 +1,80 @@
+import numpy as np
+import pytest
+from numpy.testing import assert_array_almost_equal
+
+from pylops.signalprocessing import DCT
+from pylops.utils import dottest
+
+par1 = {"ny": 11, "nx": 11, "imag": 0, "dtype": "float64"}
+par2 = {"ny": 11, "nx": 21, "imag": 0, "dtype": "float64"}
+par3 = {"ny": 21, "nx": 21, "imag": 0, "dtype": "float64"}
+
+
+@pytest.mark.parametrize("par", [(par1), (par3)])
+def test_DCT1D(par):
+    """Dot test for Discrete Cosine Transform Operator 1D"""
+
+    t = np.arange(par["ny"])
+
+    for type in [1, 2, 3, 4]:
+        Dct = DCT(dims=(par["ny"],), type=type, dtype=par["dtype"])
+        assert dottest(Dct, par["ny"], par["ny"], rtol=1e-6, complexflag=0, verb=True)
+
+        y = Dct.H * (Dct * t)
+        assert_array_almost_equal(t, y, decimal=3)
```

Comment:
This seems like a very small number of decimals. Scipy [claims that allclose should work](https://docs.scipy.org/doc/scipy/reference/generated/scipy.fftpack.idctn.html), so maybe this can become np.testing.assert_allclose.

Reply:
added in [5b8f18d](https://github.com/PyLops/pylops/pull/480/commits/5b8f18d41e7bf974ad841cbe877a58763bae4791)

Reply:
Sorry, but I believe it should be `np.testing.assert_allclose` and not `allclose`. `allclose` will not thrown an exception and so the test will always pass.

Reply:
ah sorry, i updated it 
e0471ac